subsys/net/lib/lwm2m/lwm2m_rw_senml_cbor.c

```diff
@@ -94,45 +111,74 @@ static int put_name_r(struct lwm2m_output_context *out, struct lwm2m_obj_path *p
 	return 0;
 }
 
-static int put_begin_oi(struct lwm2m_output_context *out, struct lwm2m_obj_path *path)
+/* TODO: Strictly speaking this fxn is unnecessary but used for error checking, remove later */
+static int put_begin(struct lwm2m_output_context *out, struct lwm2m_obj_path *path)
 {
 	struct cbor_out_formatter_data *fd = LWM2M_OFD_CBOR(out);
+	int ret;
 
-	if (fd->input.rec_cnt >= CONFIG_LWM2M_RW_SENML_CBOR_RECORDS) {
-		__ASSERT_PRINT("CONFIG_LWM2M_RW_SENML_CBOR_RECORDS too small");
-		return -ENOMEM;
+	ret = put_basename(out, path);
```

Comment:
Is base name only added for begin state?

Reply:
is this operation working with composite also?

Reply:
This function is actually unnecessary at this stage but let's see if there is need for it with composite operations.